### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by jackmellis**

### CODE REVIEW

File: src/signals.js [hunk 1]
```diff
@@ -8,6 +8,6 @@ export const CLOSE = prefix + 'CLOSE';
 export const NAVIGATE = prefix + 'NAVIGATE';
 
 // Creators
-export const open = createAction(OPEN);
+export const open = createAction(OPEN, (route, params) => ({route, params}));
```
Comment (jshthornton) [inline src/signals.js]: @jackmellis this is a API breaking change. To keep it from breaking over projects for no reason then keep the signature the same. Aka, take the route and params in as an object.

File: src/components/ModalRouter.js [hunk 2]
```diff
@@ -7,32 +7,36 @@ export const PureModalRouter = ({
   modal,
   modalComponent,
   closeModal,
+  ...rest,
 }) => {
   if(config == null) {
-    return cloneElement(modalComponent, {
+    return createElement(modalComponent, {
```
Comment (jshthornton) [inline src/components/ModalRouter.js]: Clone element is the right function to use. You use it something like this:
```js
<ModalRouter
  modalComponent={(<MyCustomModal/>)}
/>
```

**Comment by jackmellis** [comment]
Thing is, this library has breaking changes when compared to frieza, the signals are different, the component is not set up the same way, I couldn't find any way to get the current modal route's title and other meta data into ModalHeader via cloneElement, and this also requires passing in the modal content as Jsx when setting up modal routes (rather than just the component's render fn).

I thought it'd be easier to add breaking changes here (perhaps with a major version change) sp the usage was closer to what we had before.

**Comment by jshthornton** [comment]
Yup, I understand the idea behind it. The current style in this project has been proven that it works on other applications. I'll pull together some basic documentation of intended use

**Pull request closed without merging**